### ISSUE TITLE
Small memory leak in softmax_layer

### ISSUE THREAD

**Issue by Napilnic** [AlexeyAB/darknet#1835]
Memory is allocated for **loss** in softmax_layer.c
```
softmax_layer make_softmax_layer(int batch, int inputs, int groups)
{
...
39    l.loss = calloc(inputs*batch, sizeof(float));
...
}
```
but it isn't freed in layer.c

**Comment by AlexeyAB** [comment]
Thanks! I fixed it.
